src/components/Buttons/LinkIcon/MenuLink.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { ComponentPropsWithoutRef, useState } from "react";
+
+export interface MenuLinkProps extends ComponentPropsWithoutRef<"button"> {
+  children?: React.ReactNode;
+  icon?: JSX.Element;
+  color?: string;
+  onClick?: () => void; // Add onClick prop of type function that doesn't return anything
```

Comment:
Not sure we need that... if we use ...props we let that for the parent component. Isn't it!? Check how we did in button component.